lib/Sema/TypeCheckGeneric.cpp

```diff
@@ -576,6 +576,12 @@ static unsigned getExtendedTypeGenericDepth(ExtensionDecl *ext) {
   return sig->getGenericParams().back()->getDepth();
 }
 
+llvm::Expected<unsigned>
+GenericTypeParamDepthRequest::evaluate(Evaluator &evaluator,
+                                       GenericTypeParamDecl *gpDecl) const {
+  return gpDecl->getDeclContext()->getGenericContextDepth();
```

Comment:
IMO, the DeclContext method should be the request, not the one in GenericTypeParamDecl; that way we only cache it once.

Reply:
Let’s do both.  I’ll get that in tomorrow. This’ll fail CI anyways without Davide’s fix.